File: cpp/se/ArDomainPolicyDeleteCommand.cpp
#include "se/ArDomainPolicyDeleteCommand.hpp"
#include "se/ArExtension.hpp"
#include "se/ArDomainObjectType.hpp"
#include "se/CommandType.hpp"
#include "se/Extension.hpp"
#include "xml/XMLHelper.hpp"

#include <string>

namespace {
    class ArDomainPolicyDeleteCommandType : public CommandType
    {
        public:
            ArDomainPolicyDeleteCommandType() : CommandType(getCommandName()) { }
            std::string getCommandName() const { return "policyDelete"; }
            std::string toString() const { return "policyDelete"; }
    };

    Extension& arExtension() {
        static Extension* arExt = new ArExtension();
        return *arExt;
    }

    const ArDomainPolicyDeleteCommandType polDeleteCmdType;
    const ArDomainObjectType ardomType;
}; // anonymous namespace

ArDomainPolicyDeleteCommand::ArDomainPolicyDeleteCommand (
        const std::string &name, const std::string &reason) : ProtocolExtensionCommand(
            &polDeleteCmdType, &ardomType, name, arExtension())
{
    XMLHelper::setTextContent(
            xmlWriter->appendChild(objElement, "reason"), reason);
}

File: src/main/java/com/ausregistry/jtoolkit2/se/ArDomainPolicyDeleteCommand.java
package com.ausregistry.jtoolkit2.se;

import com.ausregistry.jtoolkit2.ErrorPkg;

/**
 * Mapping of EPP urn:ar:params:xml:ns:arext-1.0 policyDelete command specified
 * by the AusRegistry EPP extensions document.  This should be used to delete
 * domains violating relevant policy, rather than at the request of the
 * registrant.
 * Use this class to generate an AusRegistry-compliant XML document, given
 * simple input parameters.  The toXML method in Command serialises this object
 * to XML.
 */
public class ArDomainPolicyDeleteCommand extends ProtocolExtensionCommand {
    private static final long serialVersionUID = 3008976308409618621L;

    private static final CommandType POLICY_DELETE_COMMAND_TYPE = new ArPolicyDeleteCommandType();

    /**
     * @param name Required.
     * @param reason Required.
     * @throws IllegalArgumentException if {@code name} or {@code reason} is {@code null}.
     */
    public ArDomainPolicyDeleteCommand(String name, String reason) {
        super(POLICY_DELETE_COMMAND_TYPE, ExtendedObjectType.AR_DOMAIN, name);

        if (name == null || reason == null) {
            throw new IllegalArgumentException(
                    ErrorPkg.getMessage("se.ar.policyDelete.missing_arg"));
        }

        xmlWriter.appendChild(objElement, "reason").setTextContent(reason);
    }

    protected Extension getExtension() {
        return ExtensionImpl.AR;
    }
}

